misk/src/main/kotlin/misk/hibernate/Query.kt

```diff
@@ -0,0 +1,24 @@
+package misk.hibernate
+
+import kotlin.reflect.KClass
+
+/** Base class for SQL queries. */
+interface Query<T> {
+  fun uniqueResult(session: Session): T?
+  fun <P : Projection> uniqueResultAs(session: Session, projection: KClass<P>): P?
```

Comment:
should there be a method that does not return a nullable? I think a common pattern is just throwing a not found exception

Reply:
I’m trying to make it as easy as possible to migrate from Service Container, which returns null if there’s nothing found.

Reply:
`!!` is good syntax to say “this is definitely here”

Reply:
There's been plenty of bugs where we've just assumed we get something out of `uniqueResult` so I think having a reminder to think "what if this is not here?" is a good thing.